Firmware/mmu2_marlin1.cpp

```diff
@@ -0,0 +1,123 @@
+/// @file
+/// MK3 / Marlin1 implementation of support routines for the MMU2
+#include "mmu2_marlin.h"
+#include "Marlin.h"
+#include "stepper.h"
+#include "planner.h"
+#include "mmu2_config.h"
+#include "temperature.h"
+
+namespace MMU2 {
+
+void MoveE(float delta, float feedRate) {
+    current_position[E_AXIS] += delta;
+    plan_buffer_line_curposXYZE(feedRate);
+}
+
+float MoveRaiseZ(float delta) {
+    return raise_z(delta);
+}
+
+void planner_synchronize() {
+    st_synchronize();
+}
+
+bool planner_any_moves() {
+    return blocks_queued();
+}
+
+float planner_get_machine_position_E_mm(){
+    return current_position[E_AXIS];
+}
+
+float planner_get_current_position_E(){
+    return current_position[E_AXIS];
+}
+
+void planner_set_current_position_E(float e){
+        current_position[E_AXIS] = e;
+}
+
+void planner_line_to_current_position(float feedRate_mm_s){
+     plan_buffer_line_curposXYZE(feedRate_mm_s);
+     st_synchronize();
+}
+
+pos3d planner_current_position(){
+    return pos3d(current_position[X_AXIS], current_position[Y_AXIS], current_position[Z_AXIS]);
+}
+
+void motion_do_blocking_move_to_xy(float rx, float ry, float feedRate_mm_s){
+        current_position[X_AXIS] = rx;
+        current_position[Y_AXIS] = ry;
+        plan_buffer_line_curposXYZE(feedRate_mm_s);
+        st_synchronize();
+}
+
+void motion_do_blocking_move_to_z(float z, float feedRate_mm_s){
+        current_position[Z_AXIS] = z;
+        plan_buffer_line_curposXYZE(feedRate_mm_s);
+        st_synchronize();
+}
+
+void nozzle_park() {
+    current_position[X_AXIS] = MMU_ERR_X_PAUSE_POS;
+    current_position[Y_AXIS] = MMU_ERR_Y_PAUSE_POS;
+    planner_line_to_current_position(NOZZLE_PARK_XY_FEEDRATE);
+    st_synchronize();
```

Comment:
`st_synchronize()` is already included with `planner_line_to_current_position()`.

Reply:
Should maybe `plan_buffer_line_curposXYZE` be used here?